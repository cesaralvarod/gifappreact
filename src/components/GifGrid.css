dnyiv.list-items {
  padding-top: 3rem;
}

ul {
  display: grid;
  grid-template-columns: 1fr;
  gap: 1rem;
  padding: 0;
  margin: 0;
  padding-top: 1rem;
  width: 100%;
}

@media (min-width: 480px) {
  ul {
    grid-template-columns: 1fr 1fr;
  }
}

@media (min-width: 720px) {
  ul {
    grid-template-columns: repeat(3, 1fr);
  }
}

@media (min-width: 900px) {
  ul {
    grid-template-columns: repeat(4, 1fr);
  }
}
