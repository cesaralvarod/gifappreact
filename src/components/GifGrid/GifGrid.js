import React from "react";

// Components

import ListItems from "./ListItems/ListItems";

function GifGrid(props) {
  return (
    <div className="content">
      <p>Resultados de la busqueda {props.query}</p>
      <ListItems query={props.query} />
    </div>
  );
}

export default GifGrid;
