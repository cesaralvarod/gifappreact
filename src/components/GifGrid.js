import React from "react";
import PropTypes from "prop-types";

// Styles

import "./GifGrid.css";

// Components

import GifGridItem from "./GifGridItem";
import Loading from "./Loading";

// Hook

import useFetchGif from "../hooks/useFetchGifs";

function GifGrid({ query }) {
  const { data: images, loading } = useFetchGif(query);

  return (
    <div className="list-items">
      {loading ? <Loading /> : null}
      <ul>
        {images.map((img) => {
          return (
            <GifGridItem
              key={img.id}
              id={img.id}
              title={img.title}
              url={img.url}
            />
          );
        })}
      </ul>
    </div>
  );
}

GifGrid.propTypes = {
  query: PropTypes.string.isRequired,
};

export default GifGrid;
