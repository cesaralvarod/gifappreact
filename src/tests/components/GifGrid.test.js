import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import useFetchGif from "../../hooks/useFetchGifs";

// mock sirve para fingir culaquier llamada a ese archivo y suponer o controlar
// la informacion que eso va a responder
jest.mock("../../hooks/useFetchGifs");

describe("Testing in GifGrid.js", () => {
  const query = "Breaking bad";

  test("Component <GifGrid /> should show correctly", () => {
    const initUseFetchGif = { data: [], loading: true };

    useFetchGif.mockReturnValue(initUseFetchGif);

    let wrapper = shallow(<GifGrid query={query} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Component <GifGrid /> should show items when images are loaded", () => {
    let gifs = [
      {
        id: "12312sa",
        url: "https://gifurl.com",
        title: "GIF",
      },
    ];

    let expectedUseFetchGif = {
      data: gifs,
      loading: false,
    };

    useFetchGif.mockReturnValue(expectedUseFetchGif);

    const wrapper = shallow(<GifGrid query={query} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("Loading").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
