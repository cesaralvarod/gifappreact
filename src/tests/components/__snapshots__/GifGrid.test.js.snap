// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in GifGrid.js Component <GifGrid /> should show correctly 1`] = `
<div
  className="list-items"
>
  <Loading />
  <ul />
</div>
`;

exports[`Testing in GifGrid.js Component <GifGrid /> should show items when images are loaded 1`] = `
<div
  className="list-items"
>
  <ul>
    <GifGridItem
      id="12312sa"
      key="12312sa"
      title="GIF"
      url="https://gifurl.com"
    />
  </ul>
</div>
`;
